nesta_ds_utils/file_ops.py

```diff
@@ -1,53 +1,431 @@
+
+    Args:
+        list_data (list): List to convert.
+        path_to (str): Saving file name.
+
+    Returns:
+        io.BytesIO: Bytes file object.
+    """
+    buffer = io.BytesIO()
+    if fnmatch(path_to, "*.csv"):
+        pd.DataFrame(list_data).to_csv(buffer, **kwargs)
+    elif fnmatch(path_to, "*.txt"):
+        for row in list_data:
+            buffer.write(bytes(str(row) + "\n", "utf-8"))
+    elif fnmatch(path_to, "*.json"):
+        buffer.write(json.dumps(list_data, **kwargs).encode())
+    else:
+        raise Exception(
+            "Uploading list currently supported only for 'csv', 'txt' and 'json'."
+        )
+    buffer.seek(0)
+    return buffer
+
+
+def _str_to_fileobj(str_data: str, path_to: str, **kwargs) -> io.BytesIO:
+    """Convert str into bytes file object.
+
+    Args:
+        str_data (str): String to convert.
+        path_to (str): Saving file name.
+
+    Returns:
+        io.BytesIO: Bytes file object.
+    """
+    if fnmatch(path_to, "*.txt"):
+        buffer = io.BytesIO(bytes(str_data.encode("utf-8")))
+    else:
+        raise Exception("Uploading string currently supported only for 'txt'.")
+    buffer.seek(0)
+    return buffer
+
+
+def _np_array_to_fileobj(
+    np_array_data: np.ndarray, path_to: str, **kwargs
+) -> io.BytesIO:
+    """Convert numpy array into bytes file object.
+
+    Args:
+        np_array_data (np.ndarray): Numpy array to convert.
+        path_to (str): Saving file name.
+
+    Returns:
+        io.BytesIO: Bytes file object.
+    """
+    buffer = io.BytesIO()
+    if fnmatch(path_to, "*.csv"):
+        np.savetxt(buffer, np_array_data, delimiter=",", **kwargs)
+    elif fnmatch(path_to, "*.parquet"):
+        pq.write_table(pa.table({"data": np_array_data}), buffer, **kwargs)
+    else:
+        raise Exception(
+            "Uploading numpy array currently supported only for 'csv' and 'parquet."
+        )
+    buffer.seek(0)
+    return buffer
+
+
+def _unsupp_data_to_fileobj(data: any, path_to: str, **kwargs) -> io.BytesIO:
+    """Convert data into bytes file object using pickle file type.
+
+    Args:
+        data (any): Data to convert.
+        path_to (str): Saving file name.
+
+    Returns:
+        io.BytesIO: Bytes file object.
+    """
+    buffer = io.BytesIO()
+    if fnmatch(path_to, "*.pkl"):
+        pickle.dump(data, buffer, **kwargs)
+    else:
+        raise Exception(
+            "This file type is not supported for this data. Use 'pkl' instead."
+        )
+    buffer.seek(0)
+    return buffer
+
+
+def upload_obj(
+    obj: any,
+    bucket: str,
+    path_to: str,
+    kwargs_boto: dict = {},
+    kwargs_writing: dict = {},
+):
+    """Uploads data from memory to S3 location.
+
+    Args:
+        obj (any): Data to upload.
+        bucket (str): Bucket's name.
+        path_to (str): Path location to save data.
+        kwargs_boto (dict, optional): Dictionary of kwargs for boto3 function 'upload_fileobj'.
```

Comment:
I'm not 100% sure how the rendering in the docs work, but do you get nicer formatting if you use backticks for mentioning parameters, functions etc.?